### PR TITLE
TE-9603 POC Code cleanup

### PR DESCRIPTION
**Pull request by yaroslav-spryker**
Ticket: https://spryker.atlassian.net/browse/TE-9603

### CODE REVIEW

File: src/SprykerSdk/Shared/Transfer/ApplicationTransfer.php [hunk 1]
```diff
@@ -1,7 +1,334 @@
...
+/**
+ * (c) Spryker Systems GmbH copyright protected
+ */
+
 namespace SprykerSdk\Shared\Transfer;
 
-class ApplicationTransfer
+use InvalidArgumentException;
+
+class ApplicationTransfer extends AbstractTransfer
```
Comment (dereuromark) [inline src/SprykerSdk/Shared/Transfer/ApplicationTransfer.php:11]: I do not think that is a good idea
How are those updated? Manually?

Comment (yaroslav-spryker) [reply src/SprykerSdk/Shared/Transfer/ApplicationTransfer.php:11]: Yes basically what I did is just copied our Transfers and got rid of usage of Transfer`s module functionality.
From my perspective this is an easiest way to keep the code using DTO pattern.

Comment (dereuromark) [reply src/SprykerSdk/Shared/Transfer/ApplicationTransfer.php:11]: I dont think this is a good long term strategy..
It wont update with new fields.
Instead a different way of handling should be added IMO, but maybe that can be a follow up.

File: src/SprykerSdk/Integrator/IntegratorConfig.php [hunk 2]
```diff
@@ -7,13 +7,11 @@
 
 declare(strict_types = 1);
 
-namespace SprykerSdk\Zed\Integrator;
+namespace SprykerSdk\Integrator;
 
-use Spryker\Shared\Config\Config;
-use Spryker\Shared\Kernel\KernelConstants;
-use Spryker\Zed\Kernel\AbstractBundleConfig;
+use SprykerSdk\Shared\Common\AbstractConfig;
```
Comment (gechetspr) [inline src/SprykerSdk/Integrator/IntegratorConfig.php:12]: please check all such namespaces. It MUST has library name in it

File: src/SprykerSdk/Integrator/IntegratorConfig.php [hunk 3]
```diff
@@ -24,14 +22,37 @@ class IntegratorConfig extends AbstractBundleConfig
...
+     * @var array|null
+     */
+    protected $config;
+
+    /**
+     * @return void
+     */
+    public function loadConfig(): void
+    {
+        if ($this->config === null) {
```
Comment (gechetspr) [inline src/SprykerSdk/Integrator/IntegratorConfig.php:38]: early return is more preferable

File: src/SprykerSdk/Integrator/IntegratorConfig.php [hunk 4]
```diff
@@ -24,14 +22,37 @@ class IntegratorConfig extends AbstractBundleConfig
...
+            $fileName = $this->getConfigPath();
+            if (file_exists($fileName)) {
+                include $fileName;
+            }
+
+            $this->config = $this->getConfigVariableName();
```
Comment (gechetspr) [inline src/SprykerSdk/Integrator/IntegratorConfig.php:44]: is it working?

File: src/SprykerSdk/Integrator/IntegratorConfig.php [hunk 5]
```diff
@@ -24,14 +22,37 @@ class IntegratorConfig extends AbstractBundleConfig
...
+
+    /**
+     * @return void
+     */
+    public function loadConfig(): void
+    {
+        if ($this->config === null) {
+            $fileName = $this->getConfigPath();
+            if (file_exists($fileName)) {
+                include $fileName;
```
Comment (gechetspr) [inline src/SprykerSdk/Integrator/IntegratorConfig.php:41]: The idea was to merge default Spryker config with the dedicated integrator config. But currently we just get one of them. Please fix this

File: src/SprykerSdk/Integrator/IntegratorConfig.php [hunk 6]
```diff
@@ -111,7 +132,21 @@ public function getSharedConfigVariableName(): string
      *
      * @return string
      */
-    public function getSharedConfigPath(): string
+    public function getConfigPath(): string
```
Comment (gechetspr) [inline src/SprykerSdk/Integrator/IntegratorConfig.php:135]: also please check methods visibility. I guess there is no reason to keep this method public

File: src/SprykerSdk/Integrator/IntegratorFactory.php [hunk 7]
```diff
@@ -1,56 +1,62 @@
...
+use SprykerSdk\Integrator\Business\ManifestStrategy\UnwireGlueRelationshipManifestStrategy;
+use SprykerSdk\Integrator\Business\ManifestStrategy\UnwirePluginManifestStrategy;
+use SprykerSdk\Integrator\Business\ManifestStrategy\UnwireWidgetManifestStrategy;
+use SprykerSdk\Integrator\Business\ManifestStrategy\WireGlueRelationshipManifestStrategy;
+use SprykerSdk\Integrator\Business\ManifestStrategy\WirePluginManifestStrategy;
+use SprykerSdk\Integrator\Business\ManifestStrategy\WireWidgetManifestStrategy;
+use SprykerSdk\Integrator\Business\SprykerLock\SprykerLockReader;
+use SprykerSdk\Integrator\Business\SprykerLock\SprykerLockWriter;
+use SprykerSdk\Integrator\IntegratorConfig;
+use SprykerSdk\ModuleFinder\Business\ModuleFinderFacade;
```
Comment (gechetspr) [inline src/SprykerSdk/Integrator/IntegratorFactory.php:42]: Is it separate library? If no, fix namespaces

File: src/SprykerSdk/Integrator/IntegratorFactory.php [hunk 8]
```diff
@@ -1,56 +1,62 @@
...
+class IntegratorFactory
 {
     /**
-     * @return \SprykerSdk\Zed\Integrator\Business\Executor\ManifestExecutor
+     * @return \SprykerSdk\Integrator\IntegratorConfig
+     */
+    protected function getConfig(): IntegratorConfig
+    {
+        $config = IntegratorConfig::getInstance();
+        $config->loadConfig();
```
Comment (gechetspr) [inline src/SprykerSdk/Integrator/IntegratorFactory.php:53]: I assume we can load config on creating instance fase

File: src/SprykerSdk/Integrator/IntegratorFactory.php [hunk 9]
```diff
@@ -74,39 +80,39 @@ public function creatManifestExecutor(): ManifestExecutor
     }
 
     /**
-     * @return \SprykerSdk\Zed\Integrator\Business\SprykerLock\SprykerLockReader
+     * @return \SprykerSdk\Integrator\Business\SprykerLock\SprykerLockReader
      */
     public function createSprykerLockReader(): SprykerLockReader
```
Comment (gechetspr) [inline src/SprykerSdk/Integrator/IntegratorFactory.php:85]: I would like to see interfaces as return types

File: src/SprykerSdk/ModuleFinder/Business/ModuleFinderFacade.php [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+use SprykerSdk\Shared\ModuleFinder\ModuleFinderFactoryAwareTrait;
+
+class ModuleFinderFacade extends AbstractFacade implements ModuleFinderFacadeInterface
+{
+    use ModuleFinderFactoryAwareTrait;
+
+    /**
+     * {@inheritDoc}
+     *
+     * @api
```
Comment (gechetspr) [inline src/SprykerSdk/ModuleFinder/Business/ModuleFinderFacade.php:21]: remove all those tags

File: src/SprykerSdk/ModuleFinder/ModuleFinderConfig.php [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\ModuleFinder;
+
+use SprykerSdk\Shared\Common\AbstractConfig;
+
+class ModuleFinderConfig extends AbstractConfig
```
Comment (gechetspr) [inline src/SprykerSdk/ModuleFinder/ModuleFinderConfig.php:12]: this should go to the integrator config

File: src/SprykerSdk/Shared/Integrator/IntegratorFacadeAwareTrait.php [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+namespace SprykerSdk\Shared\Integrator;
+
+use SprykerSdk\Integrator\Business\IntegratorFacade;
+
+trait IntegratorFacadeAwareTrait
+{
+    /**
+     * @return \SprykerSdk\Integrator\Business\IntegratorFacade
+     */
+    protected function getFacade(): IntegratorFacade
```
Comment (gechetspr) [inline src/SprykerSdk/Shared/Integrator/IntegratorFacadeAwareTrait.php:14]: interface

**Review by gechetspr — CHANGES_REQUESTED** [review]

**Comment by gechetspr** [comment]
Also on the next iteration, please make PR against master. I want to check all files.

File: src/SprykerSdk/Integrator/IntegratorConfig.php (hunk 6 above)
Comment (geega) [reply src/SprykerSdk/Integrator/IntegratorConfig.php:135]: this method used in ConfigureEnvManifestStrategy

**Comment by dereuromark** [comment]
whats the status on this?

**Comment by gechetspr** [comment]
We can close this one. As I understand he re-created PR against master and forget about this one

**Pull request closed without merging**